api/server/middleware_auth.go

```diff
@@ -358,23 +360,22 @@ func (a *authMiddleware) enumerateWithAuth(w http.ResponseWriter, r *http.Reques
 	volIDs, ok := r.URL.Query()[api.OptVolumeID]
 	if !ok || len(volIDs[0]) < 1 {
 		a.log("", fn).Error("Failed to parse VolumeID")
+		json.NewEncoder(w).Encode(emptyVols)
 		return
 	}
 	volumeID := volIDs[0]
 
 	vols, err := d.Inspect([]string{volumeID})
 	if err != nil || len(vols) == 0 || vols[0] == nil {
 		a.log(volumeID, fn).WithError(err).Error("Failed to get volume object")
-		next(w, r)
+		json.NewEncoder(w).Encode(emptyVols)
 		return
 	}
 
-	volumeResponse := &api.VolumeResponse{}
 	tokenSecretContext, err := a.parseSecret(vols[0].Spec.VolumeLabels, vols[0].Locator.VolumeLabels, false)
 	if err != nil {
 		a.log(volumeID, fn).WithError(err).Error("failed to parse secret")
-		volumeResponse.Error = "failed to parse secret: " + err.Error()
-		json.NewEncoder(w).Encode(volumeResponse)
+		json.NewEncoder(w).Encode(emptyVols)
```

Comment:
Why are we changing the behavior here? Shouldn't we return an error to the user if he has an incorrect token/secret ?

Reply:
This is the second bug. The client gets a `inspect volume: json: cannot unmarshal object into Go value of type []*api.Volume` because the client for `Inspect()` calls the server enumerate and doesn't expect an error.

Reply:
But this is not Inspect() but it is parsing an already inspected volume and it is unable to parse the secret from the labels of that volume.